crates/ty_python_semantic/resources/mdtest/annotations/self.md

```diff
@@ -239,4 +237,32 @@ reveal_type(D().instance_method)
 reveal_type(D.class_method)
 ```
 
+In nested functions `self` binds to the method. So in the following example the `self` in `C.b` is
+bound at `C.f`.
+
+```py
+from typing import Self
+from ty_extensions import generic_context
+
+class C[T]():
+    def f(self: Self):
+        def b(x: Self):
+            reveal_type(x)  # revealed: Self@f
+        reveal_type(generic_context(b))  # revealed: None
```

Comment:
This looks right. Can we add a check that `generic_context(C.f)` contains `Self`?